codegen broken by `BinaryHeap` in metadata

The latest metadata in polkadot contains BinaryHeap which seems not be supported by subxt (this was added in scale_info v2.11)

```
Message:  Unknown prelude type 'BinaryHeap'
Location: path/.cargo/registry/src/index.crates.io-6f17d22bba15001f/scale-typegen-0.2.0/src/typegen/type_path.rs:210

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

#### Reproducible steps:

```bash
1. git clone https://github.com/paritytech/polkadot-sdk && cd polkadot-sdk && git checkout d0ebb850ed2cefeb3e4ef8b8e0a16eb7fb6b3f3e
2. cargo run -p polkadot --release
# In another terminal
3. subxt metadata --version latest -f bytes > artifacts/metadata.scale
4. subxt codegen --file artifacts/metadata.scale | rustfmt > code.rs
```

I think we have make another release of subxt-cli to bump `scale-typegen` to support the latest version of scale-info

I am having some trouble with this issue still (I think it may be the same issue). The codegen works after upgrading to 0.35.2 but using the generated code gives me errors related to the BinaryHeap (when importing the generated code here: ):

```
error[E0277]: the trait bound `EnqueuedOrder: std::cmp::Ord` is not satisfied
   --> src/parachains/relaychain.rs:1:1415422
    |
1   | ... > { :: subxt :: storage :: address :: Address :: new_static ("OnDemandAssignmentProvider" , "FreeEntries" , () , [229u8 , 190u8 ,...
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::cmp::Ord` is not implemented for `EnqueuedOrder`
    |
    = help: the trait `DecodeWithMetadata` is implemented for `DecodedValueThunk`
    = note: required for `BinaryHeap<EnqueuedOrder>` to implement `IntoVisitor`
    = note: required for `BinaryHeap<EnqueuedOrder>` to implement `DecodeAsType`
    = note: required for `BinaryHeap<EnqueuedOrder>` to implement `DecodeWithMetadata`
note: required by a bound in `subxt::storage::Address::<Keys, ReturnTy, Fetchable, Defaultable, Iterable>::new_static`
   --> /Users/claravanstaden/IdeaProjects/snowbridge/.cargo/registry/src/index.crates.io-6f17d22bba15001f/subxt-0.35.2/src/storage/storage_address.rs:95:15
    |
95  |     ReturnTy: DecodeWithMetadata,
    |               ^^^^^^^^^^^^^^^^^^ required by this bound in `Address::<Keys, ReturnTy, Fetchable, Defaultable, Iterable>::new_static`
...
100 |     pub fn new_static(
    |            ---------- required by a bound in this associated function


```

`relaychain.rs` is generated with: `subxt codegen --url ws://localhost:9944 >src/parachains/relaychain.rs` 
Running against the Rococo config on https://github.com/paritytech/polkadot-sdk master branch. Please let me know what you think the issue may be. Perhaps it is a problem in my upgrade from subxt `0.33.0` to `0.35.2`, but since the error originates from the generated code I think it may be a bug.

--- a/codegen/src/lib.rs
+++ b/codegen/src/lib.rs
@@ -391,7 +391,7 @@ fn default_substitutes(crate_path: &syn::Path) -> TypeSubstitutes {
         ),
         (
             parse_quote!(BinaryHeap),
-            parse_quote!(#crate_path::utils::KeyedVec),
+            parse_quote!(#crate_path::alloc::vec::Vec),
         ),
         (
             parse_quote!(BTreeSet),

--- a/codegen/src/lib.rs
+++ b/codegen/src/lib.rs
@@ -348,7 +348,7 @@ fn default_derives(crate_path: &syn::Path) -> DerivesRegistry {
 fn default_substitutes(crate_path: &syn::Path) -> TypeSubstitutes {
     let mut type_substitutes = TypeSubstitutes::new();
 
-    let defaults: [(syn::Path, syn::Path); 11] = [
+    let defaults: [(syn::Path, syn::Path); 12] = [
         (
             parse_quote!(bitvec::order::Lsb0),
             parse_quote!(#crate_path::utils::bits::Lsb0),
@@ -387,6 +387,10 @@ fn default_substitutes(crate_path: &syn::Path) -> TypeSubstitutes {
         // suitable type params) avoids these issues.
         (
             parse_quote!(BTreeMap),
+            parse_quote!(#crate_path::utils::KeyedVec),
+        ),
+        (
+            parse_quote!(BinaryHeap),
             parse_quote!(#crate_path::utils::KeyedVec),
         ),
         (

